src/txParsers.ts

```diff
@@ -241,49 +258,95 @@ export const parseCertificates = createParser(parseArray, parseCertificate, Pars
+        metadataHash: parseOptional(unparsedTxBody.get(TransactionBodyKeys.METADATA_HASH), parseMetadataHash),
+        validityIntervalStart: parseOptional(unparsedTxBody.get(TransactionBodyKeys.VALIDITY_INTERVAL_START), parseValidityIntervalStart),
+        mint: parseOptional(unparsedTxBody.get(TransactionBodyKeys.MINT), parseMint),
+        scriptDataHash: parseOptional(unparsedTxBody.get(TransactionBodyKeys.SCRIPT_DATA_HASH), parseScriptDataHash),
+        collaterals: parseOptional(unparsedTxBody.get(TransactionBodyKeys.COLLATERAL_INPUTS), parseCollaterals),
+        requiredSigners: parseOptional(unparsedTxBody.get(TransactionBodyKeys.REQUIRED_SIGNERS), parseRequiredSigners),
+        networkId: parseOptional(unparsedTxBody.get(TransactionBodyKeys.NETWORK_ID), parseNetworkId),
     }
 }
 
 export const parseTx = (unparsedTx: unknown): Transaction => {
-    const [body, witnessSet, auxiliaryData] = parseTuple(
+    const [body, ...otherItems] = parseTuple(
         unparsedTx,
         ParseErrorReason.INVALID_TX_CBOR,
         parseTxBody,
-        dontParse,
-        dontParse,
+        //             | shelley era:  | alonzo era:
+        dontParse,  // | witnessSet    | witnessSet
+        dontParse,  // | auxiliaryData | scriptValidity
+        dontParse,  // | `undefined`   | auxiliaryData
     )
+    const presentItems = otherItems.filter((item) => item !== undefined)
+
+    // cardano-cli with --shelley-era, --allegra-era and --mary-era
+    // includes only txBody, witnessSet and auxiliaryData
+    if (presentItems.length === 2) {
+        return { body, witnessSet: presentItems[0], auxiliaryData: presentItems[1] }
+    }
 
-    return {body, witnessSet, auxiliaryData}
+    // cardano-cli with --alonzo-era includes all tx items
+    return {
+        body,
+        witnessSet: presentItems[0],
+        scriptValidity: presentItems[1],
+        auxiliaryData: presentItems[2],
+    }
 }
 
 export const parseRawTx = (unparsedRawTx: unknown): RawTransaction => {
-    const [body, item1, item2] = parseTuple(
+    const [body, ...otherItems] = parseTuple(
         unparsedRawTx,
         ParseErrorReason.INVALID_RAW_TX_CBOR,
         parseTxBody,
-        dontParse,
-        dontParse,
+        //             | old cli:      | shelley era:    | alonzo era:
+        dontParse,  // | auxiliaryData | scriptWitnesses | scriptWitnesses
+        dontParse,  // | `undefined`   | auxiliaryData   | datumWitnesses
+        dontParse,  // | `undefined`   | `undefined`     | redeemerWitnesses
+        dontParse,  // | `undefined`   | `undefined`     | scriptValidity
+        dontParse,  // | `undefined`   | `undefined`     | auxiliaryData
     )
-    // older versions of cardano-cli did not include scriptWitnesses
-    if (item2 === undefined) {
-        return {body, scriptWitnesses: undefined, auxiliaryData: item1}
+    const presentItems = otherItems.filter((item) => item !== undefined)
```

Comment:
As above, let's check the content of the array before removing.

Reply:
https://github.com/vacuumlabs/cardano-hw-interop-lib/pull/3/commits/f7b82cf0005d25fabe891e52116525a44f45d9b4